shell.nix

```diff
@@ -39,83 +38,32 @@ let
-  # benchmarks
-  #   ocaml configure, etc. This is just source though.
-  # C deps
-  #   python headers for bootstrapping
-  # big one: Clang for ASAN and other sanitizers (even though GCC has some)
-  #   Clang for coverage too
-
-  # nixpkgs: busybox linux only; no smoosh
-  # could append something like: ++ lib.optionals stdenv.isLinux [ busybox ]
-  spec_tests = [ bash dash mksh zsh ];
-
-  static_analysis = [
-    mypy # This is the Python 3 version
-    (python2.withPackages (ps: with ps; [ flake8 pyannotate ]))
-    # python3Packages.black # wink wink :)
-  ];
-
-  binary = [ re2c ];
-  doctools = [ cmark ];
-  c_deps = [ readline ];
-
-  shell_deps = [
-    gawk
-    time
-    # additional command dependencies I found as I went
-    # guessing they go here...
-    # run with nix-shell --pure to make missing deps easier to find!
-    file
-    git
-    hostname
-    which
-  ];
-
-  nix_deps = [
-    nixfmt # `nixfmt shell.nix` to format in place
+# Create a shell with packages we need.
+mkShell rec {
+  # pull most deps from default.nix
+  buildInputs = with shells; [ test_bash test_dash test_mksh test_zsh ] ++ drv.buildInputs ++ [
+    nixfmt # `nixfmt *.nix` to format in place
   ];
 
-  # Create a shell with packages we need.
-in mkShell rec {
-
-  buildInputs = c_deps ++ binary ++ spec_tests ++ static_analysis ++ doctools
-    ++ shell_deps ++ nix_deps ++ doctools;
-
-  # Not sure if this is "right" (for nix, other platforms, etc.)
-  # doctools/cmark.py hardcoded /usr/local/lib/libcmark.so, and it looks
-  # like Nix has as much trouble with load_library as you have. For a
-  # "build" I think we'd use a patchPhase to replace the hard path
-  # in cmark.py with the correct one. Since we can't patch the source here
-  # I'm hacking an env in here and in cmark.py. Hopefully others will
-  # weigh in if there's a better way to handle this.
-  #
   # Note: Nix automatically adds identifiers declared here to the environment!
-  _NIX_SHELL_LIBCMARK = "${cmark}/lib/libcmark${stdenv.hostPlatform.extensions.sharedLibrary}";
+  _NIX_SHELL_LIBCMARK =
+    "${cmark}/lib/libcmark${stdenv.hostPlatform.extensions.sharedLibrary}";
 
   # Need nix to relax before it'll link against a local file.
   NIX_ENFORCE_PURITY = 0;
-
-  # Note: failing spec test with either of these LOCALE_ARCHIVE settings.
-  #
-  # $ test/spec.sh oil-options -r 0 -v
-  # if libc.fnmatch(pat_val.s, to_match):
-  #  SystemError: Invalid locale for LC_CTYPE
-  LOCALE_ARCHIVE =
-    if stdenv.isLinux then "${glibcLocales}/lib/locale/locale-archive" else "";
-
-  # a different option:
-  # LOCALE_ARCHIVE = lib.optionalString (stdenv.hostPlatform.libc == "glibc") "${glibcLocales}/lib/locale/locale-archive";
+  LANG = "en_US.UTF-8";
 
   # do setup work you want to do every time you enter the shell
   # Here are a few ideas that made sense to me:
   shellHook = ''
+    # Have to prefix test bash; there'll be 3+ on path
+    PATH="${shells.test_bash}/bin:$PATH"
```

Comment:
I started to note this last night, but decided I'd see how your path changes went first. Ultimately, I was explicitly specifying test shells because there's more than one bash on path, but only one is *intended* for running tests.

I don't really expect it to cause trouble, but I want to say out loud that prepending it like this *also* means the _test_ bash is the bash used for env bash shebangs while in nix-shell.

Reply:
Yes that's a good point.  Well one thing we could do it rename it to `testbash` and then invoke that... should be a fairly small change.